### DISCUSSION TITLE
New UI/UX library for TFT_eSPI

### DISCUSSION THREAD

**Discussion by riacob** [Bodmer/TFT_eSPI#2741] [Compatible support libraries]
Hello everyone, I hope you're having a good day.
I'm currently working on a library (Similar to [Bodmer/TFT_eWidget](https://github.com/Bodmer/TFT_eWidget)) which I decided to title [riacob/TFTFXLib](https://github.com/riacob/TFTFXLib).
As just said, this library provides functionality like Bodmer's, with the added bonus of integrating a "state machine" where each window has its own header file, UI, and functions/callbacks.
The user can create as many windows as they need, and, in these windows, draw any of the possible widgets (i.e. buttons with icons or text, graphs..).
The library's structure is similar to the one of [Qt](https://www.qt.io/).
The library is header-only for simplicity and inline optimizations.
The goal of the library is to allow a simple-ish creation and configuration of windows defined by the user.
This is just a very brief explanation and the documentation is currently non-existent, so if anyone is willing to help (mostly on the documentation side) they are welcome to hit me up!
I have big plans for this library, and I'm determined to keep it well-mantained and functional, upgrading it with new widgets and features as needed, and squashing bugs.

**Comment by Bodmer** [comment]
Looks promising. Post back as the library development progresses.

The Arduino IDE may not recognise the library without a library.properties file.

**Comment by riacob** [comment]
I added the library.properties file, as suggested.
I also started developing a PCB for a development board based around the ESP32-WROOM and TFT LCD. [https://oshwlab.com/iacob.riccardo/esp32_tft_4inch](https://oshwlab.com/iacob.riccardo/esp32_tft_4inch).

**Comment by riacob** [comment]
Hey there,
I have been quite busy lately, but today I got some time to write down a draft for the documentation and fix some code.
Also, the development board will be ready soon. 
I have continued to test the library and there don't seem to be any major bugs, at least for the time being. 
I am however looking for some testers, maybe with different displays and micros?
